src/libcephfs_proxy/libcephfsd.c

```diff
@@ -1828,7 +1829,28 @@ static int32_t server_start(proxy_manager_t *manager)
 static void log_print(proxy_log_handler_t *handler, int32_t level, int32_t err,
 		      const char *msg)
 {
-	printf("[%d] %s\n", level, msg);
+	static const char failure[] = "[\?] (\?\?\?\?) ";
+	static const char level_chars[] = "CEWID";
+
+	char header[32];
+	struct iovec iov[3];
+	int32_t len;
+
+	len = snprintf(header, sizeof(header), "[%c] (%4d) ",
+		       level_chars[level], err);
```

Comment:
Unless it is an immediate error for a function call it might not make sense to print the error code.
```
[I] (   0) Features enabled: 00000001
[I] (   0) Features enabled: 00000001
[I] (   0) Created a new client instance (0x7fd2ec010bd0)
[I] (   0) Features enabled: 00000000
[I] (   0) Features enabled: 00000000
```

Do you still think it is useful?

Reply:
I modified the code to only log the error if it's not 0. This should cover all the cases.